components/ListView/ComponentInputs.js

```diff
@@ -19,178 +28,91 @@ const messages = defineMessages({
   }
 });
 
-class ComponentInputs extends React.Component {
-  componentDidMount() {
-    this.initializeBootstrapElements();
-    this.bindTooltipShow();
-    this.bindHideTooltip();
-    this.bindTooltipMouseleave();
-  }
-
-  componentDidUpdate() {
-    this.unbind();
-    this.initializeBootstrapElements();
-    this.bindTooltipShow();
-    this.bindHideTooltip();
-    this.bindTooltipMouseleave();
-    this.hideTooltip("all");
-  }
-
-  componentWillUnmount() {
-    this.unbind();
-    this.hideTooltip("all");
-  }
-
-  bindTooltipShow() {
-    $(".cmpsr-list-inputs")
-      .off()
-      .on("mouseenter focus", '[data-toggle="tooltip"]', event => {
-        // prevent li tooltip from flashing when focus moves to the <a>
-        event.stopPropagation();
-        // hide tooltip for other list items
-        if ($(event.currentTarget).hasClass("list-pf-container")) {
-          $('.list-pf-container[data-toggle="tooltip"]')
-            .not(event.target)
-            .tooltip("hide");
-        }
-        // hide tooltip for component list item if hovering over an action
-        if ($(event.currentTarget).parent(".list-pf-actions").length) {
-          this.hideTooltip("parent");
-        }
-        $(event.currentTarget).tooltip("show");
-      });
-  }
-
-  bindHideTooltip() {
-    $(".cmpsr-list-inputs").on("blur mousedown", '[data-toggle="tooltip"]', event => {
-      // prevent focus event so that tooltip doesn't display again on click
-      event.preventDefault();
-      this.hideTooltip(event.currentTarget);
-    });
-  }
-
-  bindTooltipMouseleave() {
-    $(".cmpsr-list-inputs").on("mouseleave", '[data-toggle="tooltip"]', event => {
-      this.hideTooltip(event.currentTarget);
-      if ($(event.currentTarget).parent(".list-pf-actions").length) {
-        $(event.currentTarget)
-          .parents(".list-pf-container")
-          .tooltip("show");
-      }
-    });
-  }
-
-  unbind() {
-    $(".list-pf-actions").off("mouseenter focus mouseleave blur mousedown");
-  }
-
-  hideTooltip(target) {
-    if (target === "all") {
-      $('.cmpsr-list-inputs [data-toggle="tooltip"][aria-describedby]').tooltip("hide");
-    } else if (target === "parent") {
-      $('.list-pf-container[data-toggle="tooltip"][aria-describedby]').tooltip("hide");
-    } else {
-      $(target).tooltip("hide");
-    }
-  }
-
-  initializeBootstrapElements() {
-    // Initialize Boostrap-tooltip
-    $('[data-toggle="tooltip"]').tooltip({
-      trigger: "manual"
-    });
-  }
-
-  handleEnterKey(event, component) {
-    if (event.which === 13 || event.keyCode === 13) {
-      const { handleComponentDetails } = this.props;
-      handleComponentDetails(event, component);
-    }
-  }
-
+class ComponentInputs extends React.PureComponent {
```

Comment:
@jkozol The tests fail because changing this component to `React.PureComponent` results in the component not re-rendering when adding/removing components. 

I can change this line to the following to fix the re-render issue, but am wondering if you have any alternate suggestions.
```
// eslint-disable-next-line react/prefer-stateless-function
class ComponentInput extends React.Component {
```

Reply:
I went ahead and pushed this change, just to see if the tests will pass.

Reply:
This fixed the issue for me and I think is good for now. I think not being able to use a PureComponent here might indicate we should reconsider how we mutate the list of available components.